cmd/memcached_exporter/main.go

```diff
@@ -50,8 +59,30 @@ func main() {
 	level.Info(logger).Log("msg", "Starting memcached_exporter", "version", version.Info())
 	level.Info(logger).Log("msg", "Build context", "context", version.BuildContext())
 
+	if *serverName == "" {
+		*serverName = strings.Split(*address, ":")[0]
```

Comment:
This will break with input like `[2001:DB8::1]:1234`. Maybe you want to use [net.SplitHostPort](https://pkg.go.dev/net#SplitHostPort)?